solr/core/src/java/org/apache/solr/update/UpdateLog.java

```diff
@@ -1940,7 +1940,12 @@ public void doReplay(TransactionLog translog) {
         ThreadLocal<UpdateRequestProcessor> procThreadLocal =
             ThreadLocal.withInitial(
                 () -> {
-                  var proc = processorChain.createProcessor(req, rsp);
+                  UpdateRequestProcessor proc;
+                  try {
+                    proc = processorChain.createProcessor(req, rsp);
+                  } catch (IOException e) {
+                    throw new RuntimeException(e);
+                  }
```

Comment:
After removing IOException from UpdateRequest files - this change isn't needed.

Reply:
Hi @risdenk 
if we see this lucene commit https://github.com/apache/lucene/pull/11998/files , the `KNearestNeighborDocumentClassifier` constructor now throws IOException 
we are using that in `ClassificationUpdateProcessor`'s constructor 
so that's why I added `throws IOExceptio`n in the `ClassificationUpdateProcessor` constructor
and we are calling `ClassificationUpdateProcessor` from ClassificationUpdateProcessorFactory#getInstance() 
so I had to throw IOException from ClassificationUpdateProcessorFactory#getInstance() 
since it is inherited from UpdateRequestProcessorFactory, I had to add it there as well.
and so on for other related files like this.

Reply:
Yea lets not propagate `IOEXception` everywhere if we don't need to.